src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -282,25 +354,29 @@ private bool ContainsKeyThemeMerged(object key)
 		private void CopyFrom(ResourceDictionary source)
 		{
 			_values.Clear();
-			MergedDictionaries.Clear();
+			_mergedDictionaries.Clear();
 			_themeDictionaries?.Clear();
 
-			_values.AddRange(source);
-			MergedDictionaries.AddRange(source.MergedDictionaries);
+			_values.AddRange(source._values);
```

Comment:
am I reading this correctly: with this change it won't eagerly materialize on copy?

Reply:
No, this change does not impact this behavior. The new internal dictionary does not expose AddRange the same way the previous one did, hence the change. Non-materializing copy is for another PR.